lib/galaxy/objectstore/__init__.py

```diff
@@ -322,7 +322,7 @@ def __init__(self, config, config_dict=None, **kwargs):
         if config_dict is None:
             config_dict = {}
         super().__init__(config=config, config_dict=config_dict, **kwargs)
-        self.store_by = config_dict.get("store_by", None) or getattr(config, "object_store_store_by", "id")
+        self.store_by = config_dict.get("store_by") or config.object_store_store_by
```

Comment:
The objectstore module is meant to be used standalone, it doesn't have direct dependencies on the config module if you look at the package structure. I prefer the getattr for those reasons. This might not even be purely stylistic, I'm not sure Pulsar for instance sets up object_store_by on its config object equivalent.

Reply:
Thank you for the explanation - I didn't know that. I'll undo this edit.